### PR TITLE
Template for spark

### PR DESCRIPTION
**Pull request by soltysh**
This is not-fully functional template. Still the variable expansion needs a bit work. Currently I'm creating randomly generated names for master and worker (so they don't interfere, both names and selectors). I'm passing the generated master name as a first argument to worker command, but it does not work properly yet.

### CODE REVIEW

**Comment by soltysh** [comment]
To test it run `oc new-app tempalate.yaml` it will process it and submit to the cluster.

File: worker/start.sh [hunk 1]
```diff
@@ -3,9 +3,10 @@
 . /common.sh
 
 # name resolution for spark-master
-echo "${SPARK_MASTER_SERVICE_HOST:-$1} spark-master" >> /tmp/hosts
+declare "service_name=$(echo $1 | awk '{print toupper($0)}')_SERVICE_HOST"
+echo "${!service_name} spark-master" >> /tmp/hosts
```
Comment (mattf) [inline worker/start.sh]: what about -

echo "$(eval echo \$${1^^}) spark-master" >> /tmp/hosts

File: worker/start.sh [hunk 2]
```diff
@@ -3,9 +3,10 @@
...
 
-# because the hostname only resolves locally
+# # because the hostname only resolves locally
```
Comment (mattf) [inline worker/start.sh]: nix this

File: worker/start.sh (hunk 1 above)
Comment (soltysh) [inline worker/start.sh]: I'll give it a try in about an hour. I'm not in front of computer atm.

**Comment by soltysh** [comment]
@mattf aside from uppercasing I needed to also change dashes to underscores. Now it's working :)

**Comment by mattf** [comment]
lgtm

**Pull request merged**